### ISSUE TITLE
do while with no curly braces on two lines is allowed when using ignore-same-line

### ISSUE THREAD

**Issue by Tyriar** [palantir/tslint#4026]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.9.2
- __Running TSLint via__: VSCode, CLI

#### TypeScript code being linted

```ts
do i++;
while (i < 10);
```

with `tslint.json` configuration:

```json
    "curly": [
      true,
      "ignore-same-line"
    ],
```

#### Actual behavior

The code is allowed.

#### Expected behavior

I would expect this to fail as it's not really a single line statement like `if (a) return;` because the `while` is on the following line. I think this is the ideal behavior:

```ts
// Fail
do i++;
while (i < 10);

// Pass
if (a) return;

// Pass
do i++; while (i < 10);
```

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
